data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -152,11 +140,45 @@ $IPTABLES -t raw -A DHCP_IRONIC -j DROP
 {{end}}
 
 # Wait for images to be downloaded/ready
-podman wait -i 1000ms ipa-downloader
 podman wait -i 1000ms coreos-downloader
-while ! curl --fail http://localhost/images/rhcos-ootpa-latest.qcow2.md5sum ; do sleep 1; done
-while ! curl --fail --head http://localhost/images/ironic-python-agent.initramfs ; do sleep 1; done
-while ! curl --fail --head http://localhost/images/ironic-python-agent.kernel ; do sleep 1; done
+
+export KUBECONFIG=/opt/openshift/auth/kubeconfig-loopback
+
+mkdir -p /tmp/nmstate
+
+{{range .PlatformData.BareMetal.Hosts}}
+    until oc get -n openshift-machine-api baremetalhost {{.Name}}; do
+        echo Waiting for Host {{.Name}} to appear...
+        sleep 10
+    done
+    secret_name=$(oc get -n openshift-machine-api baremetalhost {{.Name}} -o jsonpath="{.spec.preprovisioningNetworkDataName}")
+    if [ -n "${secret_name}" ]; then
+        until oc get -n openshift-machine-api secret "${secret_name}"; do
+            echo Waiting for Secret "${secret_name}" to appear...
+            sleep 10
+        done
+        oc get -n openshift-machine-api secret ${secret_name} -o jsonpath="{.data.nmstate}" | base64 -d > /tmp/nmstate/{{.Name}}.yaml
+    else
+        touch /tmp/nmstate/{{.Name}}.yaml
+    fi
+{{end}}
+
+
+# Embed agent ignition into the rhcos live iso
+sudo podman run -d -p 8084:8084 --privileged --name image-customization \
```

Comment:
Does it really have to be privileged?

Reply:
Based on experience with CBO, probably yes because all of the image files in the cache are owned by root. This may be fixable with careful examination of what we are running as privileged.